File: data/readable_format/cpp/file1880.cpp
#include <iostream>
#include <string>

// Function to match the Fortran behavior
std::string getCaseString(int nn) {
    switch(nn) {
        case 1:
            return "case1";
        case 2:
            return "case2";
        default:
            return "";  // No output for cases not explicitly defined, aligning with Fortran
    }
}

// Test function
void testCases() {
    std::cout << getCaseString(1) << std::endl; // Expected to print "case1"
    std::cout << getCaseString(2) << std::endl; // Expected to print "case2"
    std::cout << getCaseString(3) << std::endl; // Expected not to print anything
}

int main() {
    testCases();
    return 0;
}

File: data/readable_format/fortran/file1880.f90
program test_select_case
  integer :: nn

  ! Test cases
  nn = 1
  call testCase(nn)
  nn = 2
  call testCase(nn)
  nn = 3  ! This will test a value that doesn't match any case.
  call testCase(nn)

contains

  subroutine testCase(nn)
    integer, intent(in) :: nn
    select case(nn)
    case(1)
      write(*,*) "case1"
    case(2)
      write(*,*) "case2"
    case default
      ! Explicitly handling default case to match C++ behavior,
      ! but not producing any output to align with the initial Fortran logic.
    end select
  end subroutine testCase

end program test_select_case